filecoin-proofs/src/bin/phase2.rs

```diff
@@ -1018,6 +1022,67 @@ fn setup_logger(log_filename: &str) {
     });
 }
 
+fn parameter_identifier<Tree: 'static + MerkleTreeTrait>(sector_size: u64, proof: Proof) -> String {
+    match proof {
+        Proof::Sdr => {
+            let n_partitions = *POREP_PARTITIONS.read().unwrap().get(&sector_size).unwrap();
+
+            let porep_config = PoRepConfig {
+                sector_size: SectorSize(sector_size),
+                partitions: PoRepProofPartitions(n_partitions),
+                porep_id: [0; 32],
```

Comment:
I'm not certain that setting the porep-id to all zeroes will produce the correct identifier for mainnet params which have a nonzero porep-id. It may, but idk

Reply:
I got them from https://github.com/filecoin-project/rust-fil-proofs/blob/5e760a737ffb74267c119bc662b865b884f1aac3/filecoin-proofs/src/bin/phase2.rs#L277. I used the parameter of the initial files.

Reply:
The phase2 initial parameter gen won't be affected by the porep-id, that's why it's set to all zeros. But, I didn't know whether or not the porep-id will affect the identifier

Reply:
https://github.com/filecoin-project/rust-fil-proofs/blob/5e760a737ffb74267c119bc662b865b884f1aac3/storage-proofs/porep/src/stacked/vanilla/params.rs#L84-L91 is called on the public parameters.

Example output: `layered_drgporep::PublicParams{ graph: stacked_graph::StackedGraph{expansion_degree: 8 base_graph: drgraph::BucketGraph{size: 1073741824; degree: 6; hasher: poseidon_hasher} }, challenges: LayerChallenges { layers: 11, max_count: 18 }, tree: merkletree-poseidon_hasher-8-8-0 }`

So it doesn't look like the `porep_id` is part of it.

Reply:
It's possible that there is a bug (no reason to expect it, and a scan of code suggests not), but for reference:

- The identifier used to derive the working file for the parameters should *not* depend on `porep_id`.
- There should be a 1-1 relationship between distinct groth params and file names.
- `porep_id` is intentionally decoupled from the concrete circuit (by design, it can be used to change the graphs without changing the circuit).
- This explains both why phase2 does not require `porep_id` and also why it must not affect the generated filename.
- If the generated filename *does* depend on `porep_id`, that is a bug — and we should fix it.
- Here is the code which I *think* shows that it does not (even though it 'could'): https://github.com/filecoin-project/rust-fil-proofs/blob/master/storage-proofs/porep/src/stacked/vanilla/params.rs#L84-L91

So I think it should be safe to assume an arbitrary `porep_id` can be supplied. However, it would be worth verifying that choice of `porep_id` doesn't affect the final filenames — as an independent check on this important logic.